api/src/main/java/io/grpc/ServerCall.java

```diff
@@ -208,7 +208,20 @@ public void setMessageCompression(boolean enabled) {
    */
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1704")
   public void setCompression(String compressor) {
-    // noop
+  }
+
+  /**
+   * Returns the level of security guarantee in communications
+   *
+   * <p>Determining the level of security offered by the transport for RPCs on server-side.
+   * This can be approximated by looking for the SSLSession, but that doesn't work for ALTS and
+   * maybe some future TLS approaches. May returns a lower security level in the face of
+   * uncertainty.
+   *
+   * @return non-{@code null} SecurityLevel enum
+   */
+  public SecurityLevel getSecurityLevel() {
```

Comment:
Should be marked `@ExperimentalApi` because `SecurityLevel` is experimental (https://github.com/grpc/grpc-java/issues/4692) and ServerCall Attributes are Experimental (https://github.com/grpc/grpc-java/issues/1779).